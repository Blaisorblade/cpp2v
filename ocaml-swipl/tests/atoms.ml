open Swipl

let atom = "my_atom"

let _ =
  Embedding.initialise [|Sys.argv.(0); "-q"|];
  let atom1 = Atom.make atom in
  let atom2 = Atom.make atom in
  let node1 = Term.Ref.make () in
  let node2 = Term.Ref.make () in
  Term.put_atom node1 atom1;
  Term.put_atom node2 atom2;
  Printf.printf "Compare: %i\n%!" (Term.compare node1 node2);
  Printf.printf "Equal: %b\n%!" (Term.equal node1 node2);
  Printf.printf "Unifiable: %b\n%!" (Term.unify node1 node2)
