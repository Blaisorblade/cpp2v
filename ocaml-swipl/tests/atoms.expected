Compare: 0
Equal: true
Unifiable: true
